### DISCUSSION TITLE
Custom "Modes/Themes" (Ex. Dark Mode)

### DISCUSSION THREAD

**Discussion by xMrAfonso** [tailwindlabs/tailwindcss#6547] [Ideas]
First of all, sorry if I explain this incorrectly, but my idea is to let us customize "modes". Currently we have a built-in light & dark mode, the idea is to let us add other modes. This would help a lot with accessibility and even let us add ways to help people with color issues.
Right now we have, for example, ``dark:bg-blue-500`` and the idea is to let us add other "modes". So, if I add a mode called ``blue``, I can use ``blue:bg-blue-500``. You may even use both, and have the issue decide which theme/mode he uses.

PS: If this already exists, sorry.

**Comment by CanRau** [comment]
There might be an easier way but this worked for me, so I can change the class `<html class="blue">` and use `blue:text-red-500` on an element

```js
// tailwind.config.js

const customVariants = plugin(({ addVariant, e }) => {
  // addVariant("blue", ".blue &");
  addVariant("blue", ({ modifySelectors, separator }) => {
    modifySelectors(({ className }) => {
      const element = e(`blue${separator}${className}`);
      return `.blue .${element}`;
    });
  });
});

module.exports = {
  plugins: [
    customVariants,
  ],
  // [..] rest of your config
};
```

**Reply by xMrAfonso** [reply]
So, I would have to do this for several "themes" right?

**Reply by CanRau** [reply]
Yeah, like so

```js
// tailwind.config.js

const customVariants = plugin(({ addVariant, e }) => {
  addVariant("blue", ({ modifySelectors, separator }) => {
    modifySelectors(({ className }) => {
      const element = e(`blue${separator}${className}`);
      return `.blue .${element}`;
    });
  });
  addVariant("yellow", ({ modifySelectors, separator }) => {
    modifySelectors(({ className }) => {
      const element = e(`yellow${separator}${className}`);
      return `.yellow .${element}`;
    });
  });
});

module.exports = {
  plugins: [
    customVariants,
  ],
  // [..] rest of your config
};
```

or "automated" like so

```js
// tailwind.config.js

const variants = ["blue", "red", "yellow"];

const customVariants = plugin(({ addVariant, e }) => {
  variants.map((variant) =>
    addVariant(variant, ({ modifySelectors, separator }) => {
      modifySelectors(({ className }) => {
        const element = e(`${variant}${separator}${className}`);
        return `.${variant} .${element}`;
      });
    }),
  );
});

module.exports = {
  plugins: [customVariants],
  // [..] rest of your config
};
```

But depending on your needs, I suggest checking out what @NovusOpus wrote https://github.com/tailwindlabs/tailwindcss/discussions/6547#discussioncomment-1820585

**Comment by NovusOpus** [comment]
Check this [Screencast](https://www.youtube.com/watch?v=MAtaT8BZEAo) from Tailwind CSS Official YouTube Channel.

In this Tutorial @simonswiss, Show you how to create different themes with color variations of the same components, using CSS Variables.

And here's an example in action:[ TailwindPlay](https://play.tailwindcss.com/YelhilBeHb)
